sceptre/plan/plan.py

```diff
@@ -34,10 +34,13 @@ def __init__(self, context):
         all_stacks, command_stacks = config_reader.construct_stacks()
         self.graph = StackGraph(all_stacks)
         self.command_stacks = command_stacks
+        self.config_reader = config_reader
 
     def _execute(self, *args):
         executor = SceptrePlanExecutor(self.command, self.launch_order)
-        return executor.execute(*args)
+        result = executor.execute(*args)
+        self.config_reader.clean()
```

Comment:
[Optional] This is not a big deal but it's not a real cache if we are deleting after every run?   The files are always downloaded even if nothing changes.  It's more like just a temp location for the downloaded templates.   My idea of a cache is that files wouldn't need to be downloaded at all if the ones in the cache is the same as the one being referenced?  That would be really nice if you have lots of templates (and dependencies) to deploy or update.

Reply:
Agreed, I will call it a temp dir for now (maybe in .sceptre/tmp). The caching of files could be something for a next iteration. Time-wise I'm not sure how much we will win by caching the templates for another run, as you would have to hash the contents and check that against the MD5 of the file on S3. For saving on transferred bytes it is a good solution of course.